src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/unit-creation-card/unit-creation-card.component.ts

```diff
@@ -1,15 +1,56 @@
-import { Component } from '@angular/core';
+import { Component, EventEmitter, Input, Output } from '@angular/core';
 import { faCheck, faFileUpload, faLink, faScroll, faVideo } from '@fortawesome/free-solid-svg-icons';
+import { ActivatedRoute, Router } from '@angular/router';
+import { ArtemisNavigationUtilService } from 'app/utils/navigation.utils';
+import { LectureUnitType } from 'app/entities/lecture-unit/lectureUnit.model';
 
 @Component({
     selector: 'jhi-unit-creation-card',
     templateUrl: './unit-creation-card.component.html',
 })
 export class UnitCreationCardComponent {
+    @Input() emitEvents = false;
+
+    @Output()
+    onUnitCreationCardClicked: EventEmitter<LectureUnitType> = new EventEmitter<LectureUnitType>();
+
+    unitType = LectureUnitType;
+
     // Icons
     faCheck = faCheck;
     faVideo = faVideo;
     faFileUpload = faFileUpload;
     faScroll = faScroll;
     faLink = faLink;
+
+    constructor(protected activatedRoute: ActivatedRoute, private navigationUtilService: ArtemisNavigationUtilService, private router: Router) {}
+
+    onButtonClicked(type: LectureUnitType) {
+        if (this.emitEvents) {
+            this.onUnitCreationCardClicked.emit(type);
+            return;
+        }
+
+        let navigationTarget = [];
+
+        switch (type) {
```

Comment:
We should find an alternative approach than switch cases for this in the future. e.g. inheritance or something